near-sdk/src/environment/env.rs

```diff
@@ -279,6 +279,37 @@ pub fn keccak512_hash(value: &[u8]) -> [u8; 64] {
     }
 }
 
+/// Recovers an ECDSA signer address from a 32-byte message `hash` and a corresponding `signature`
+/// along with `v` recovery byte.
+///
+/// Takes in an additional flag to check for malleability of the signature
+/// which is generally only ideal for transactions.
+///
+/// Returns 64 bytes representing the public key if the recovery was successful.
+pub fn ecrecover(
+    hash: &[u8; 32],
+    signature: &[u8; 64],
```

Comment:
I was contemplating this too and went with statically known array sizes just to make the API more "robust", but you are right that it can be awkward to use. I don't have a strong opinion here, so let's go with your option.

Reply:
yeah, either way would be fine, esp since the array slice can be converted to slice with no breaking changes later if community needed, but I don't really see a downside to this change other than it might be unclear at a glance what sizes are expected.